ZipBuilder/Sources/FirebaseReleaser/Tags.swift

```diff
@@ -77,4 +79,27 @@ enum Tags {
       fatalError("Do not remove tag of a published Firebase version.")
     }
   }
+
+  private static func verifyTagIsSafeToAdd(_ tag: String, gitRoot: URL) {
+    if checkTagCommand("git tag -l \(tag)", gitRoot: gitRoot) != "" {
+      fatalError("Tag \(tag) already exists locally. Please delete and restart")
+    }
+    if checkTagCommand("git ls-remote origin refs/tags/\(tag)", gitRoot: gitRoot) != "" {
+      fatalError("Tag \(tag) already exists locally. Please delete and restart")
+    }
+  }
+
+  private static func checkTagCommand(_ command: String, gitRoot: URL) -> String {
```

Comment:
Could you please add some explanation to what the method does to help us in the future with understanding the code.